### PR TITLE
Added unit tests for app flow simulation

### PR DESCRIPTION
**Pull request by bonniecpk**

### CODE REVIEW

File: oauthdoctor/oauth/installed_app_test.go [hunk 1]
```diff
@@ -1,34 +1,125 @@
...
+
+	"github.com/googleads/google-ads-doctor/oauthdoctor/diag"
+	"golang.org/x/oauth2"
 )
 
+func setupFakeOAuthServer() (*httptest.Server, func()) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/auth", func(w http.ResponseWriter, r *http.Request) {
+		// Should return authorization code back to the user
+		w.Write([]byte("code=mockauthcode"))
```
Comment (bobhancock) [inline oauthdoctor/oauth/installed_app_test.go]: nit: this is a fake not a mock server, so do you want to call this a mockoauthcode?

Fake: code that implements an interface but contains fixed data and no logic. Simply returns "good" or "bad" data depending on the implementation.

Mock: code that implements an interface and allows the ability to dynamically set the values to return/exceptions to throw from particular methods and provides the ability to check if particular methods have been called/not called.

File: oauthdoctor/oauth/installed_app_test.go [hunk 2]
```diff
@@ -1,34 +1,125 @@
...
-    got := genAuthCodePrompt(tt.input)
-    if !strings.HasPrefix(got, tt.want) {
-      t.Errorf("\n%s\nprompt does not match operating system want=%s\ngot=%s\ninput=%s\n",
-        tt.desc, tt.want, got, tt.input)
-    }
-  }
+	for _, tt := range tests {
+		got := genAuthCodePrompt(tt.input)
+		if !strings.HasPrefix(got, tt.want) {
+			t.Errorf("\n%s\nprompt does not match operating system got=%s\nwant=%s\ninput=%s\n",
```
Comment (bobhancock) [inline oauthdoctor/oauth/installed_app_test.go]: The standard practice it to show what you called with arguments and then got/want.
e.g., t.Errorf("genAutCodePrompt(%s) got=%s\nwant=%s", ninput, got, tt.want)

Comment (bonniecpk) [reply oauthdoctor/oauth/installed_app_test.go]: Fixed

File: oauthdoctor/oauth/installed_app_test.go (hunk 1 above)
Comment (bonniecpk) [reply oauthdoctor/oauth/installed_app_test.go]: Fixed

**Pull request merged**